book_price/models/sale_order_line.py

```diff
@@ -0,0 +1,22 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    pricelist_price_unit = fields.Float(string="Book Price", compute='_compute_pricelist_price_unit')
+
+    @api.depends('product_id', 'product_uom', 'product_uom_qty')
+    def _compute_pricelist_price_unit(self):
+        for line in self:
+            if line.product_id:
+                line.pricelist_price_unit = line._get_pricelist_price()
+            else:
+                line.pricelist_price_unit = 0.0
+
+    def _prepare_invoice_line(self, **optional_values):
+        self.ensure_one()
+        res = super()._prepare_invoice_line(**optional_values)
+        res.update({'pricelist_price_unit': self.pricelist_price_unit})
+        return res
```

Comment:
```suggestion
        if self.pricelist_price_unit:
            res['pricelist_price_unit'] = self.pricelist_price_unit
        return res
```
as we are modifying just one field, we can directly assign it rather than using update which is generally used for batch updates.